test/metabase/query_processor/middleware/permissions_test.clj

```diff
@@ -85,6 +85,21 @@
                :type     :query
                :query    {:source-table (u/the-id table)}}))))))
 
+(deftest nested-native-query-test
```

Comment:
Accidentally removed this test in https://github.com/metabase/metabase/pull/47332 so I'm adding it back here 🤦